src/Models/HydrostaticFreeSurfaceModels/barotropic_pressure_correction.jl

```diff
@@ -0,0 +1,36 @@
+import Oceananigans.TimeSteppers: calculate_pressure_correction!, pressure_correct_velocities!
+
+calculate_pressure_correction!(::HydrostaticFreeSurfaceModel, Δt) = nothing
+
+#####
+##### Barotropic pressure correction for models with a free surface
+#####
+
+pressure_correct_velocities!(model::HydrostaticFreeSurfaceModel{T, E, A, <:ExplicitFreeSurface}, Δt) where {T, E, A} = nothing
+
+#####
+##### Barotropic pressure correction for models with a free surface
+#####
+
+function pressure_correct_velocities!(model::HydrostaticFreeSurfaceModel{T, E, A, <:ImplicitFreeSurface}, Δt) where {T, E, A}
+
+    event = launch!(model.architecture, model.grid, :xyz,
+                    _barotropic_pressure_correction,
+                    model.velocities,
+                    model.grid,
+                    Δt,
+                    model.free_surface.gravitational_acceleration,
+                    model.free_surface.η,
+                    dependencies = Event(device(model.architecture)))
+
+    return nothing
+end
+
+@kernel function _barotropic_pressure_correction(U, grid, Δt, g, η)
+    i, j, k = @index(Global, NTuple)
+
+    @inbounds begin
+        U.u[i, j, k] -= g * ∂xᶠᵃᵃ(i, j, k, grid, η)
+        U.v[i, j, k] -= g * ∂yᵃᶠᵃ(i, j, k, grid, η)
```

Comment:
@francispoulin you wrote:

> In barotropic_pressure_correction.jl, lines 33 and 34, you are computing hat looks like the geostrophic velocity but there's no Coriolis, so I guess it's not that. Is this just the pressure force due to the free surface,-g \nabla \eta? I'm trying to follow the code but a bit confused what U.u mans.

Reply:
This is a pressure correction, so `U.u` is a "predictor velocity" generated by time-stepping the velocity field _without_ the contribution from the barotropic pressure, `g * eta`.

I think we will use this only when taking an implicit time-step for the free surface.